docs/public/specifications/test-zones/Connectivity-TP/connectivity04.md

```diff
@@ -0,0 +1,295 @@
+
+
+## Background
+
+See the [test scenario README file].
+
+
+## Test Case
+This document specifies defined test scenarios for test case [Connectivity04].
+
+
+## Test scenarios
+
+The purpose of the test scenarios is to cover all reasonable contexts where
+different message tags are outputted when [Connectivity04] is run on a test zone.
+The message tags are defined in the test case ([Connectivity04]) and the
+scenarios are defined below.
+
+The test scenarios are structured as stated in the [test scenario README file].
+
+## Test zone names
+
+The test zone or zones for each test scenario in this document is a subdomain
+(or lower zone) delegated from the base name (`connectivity04.xa`) and that subdomain
+having the same name as the scenario. The names of those zones are given in
+section "[Test scenarios and setup of test zones]" below.
+
+
+## All message tags
+The test case can output any of these message tags, but not necessarily in any
+combination. See [Connectivity04] for the specification of the tags.
+
+* CN04_EMPTY_PREFIX_SET
+* CN04_ERROR_PREFIX_DATABASE
+* CN04_IPV4_DIFFERENT_PREFIX
+* CN04_IPV4_SAME_PREFIX
+* CN04_IPV4_SINGLE_PREFIX
+* CN04_IPV6_DIFFERENT_PREFIX
+* CN04_IPV6_SAME_PREFIX
+* CN04_IPV6_SINGLE_PREFIX
+
+## Test scenarios and message tags
+
+If a message tag is not listed for the scenario, its presence or non-presence is
+irrelevant to the test scenario and must be ignored.
+
+Scenario name            | Mandatory message tag                                                             | Forbidden message tags
+:------------------------|:----------------------------------------------------------------------------------|:--------------------
+GOOD-1                   | CN04_IPV4_DIFFERENT_PREFIX, CN04_IPV6_DIFFERENT_PREFIX                            | 2)
+GOOD-2                   | CN04_IPV4_DIFFERENT_PREFIX                                                        | 2)
+GOOD-3                   | CN04_IPV6_DIFFERENT_PREFIX                                                        | 2)
+EMPTY-PREFIX-SET-1       | CN04_EMPTY_PREFIX_SET                                                             | 2)
+EMPTY-PREFIX-SET-2       | CN04_EMPTY_PREFIX_SET                                                             | 2)
+ERROR-PREFIX-DATABASE-1  | CN04_ERROR_PREFIX_DATABASE                                                        | 2)
+ERROR-PREFIX-DATABASE-2  | CN04_ERROR_PREFIX_DATABASE                                                        | 2)
+ERROR-PREFIX-DATABASE-3  | CN04_ERROR_PREFIX_DATABASE                                                        | 2)
+ERROR-PREFIX-DATABASE-4  | CN04_IPV4_DIFFERENT_PREFIX, CN04_IPV6_DIFFERENT_PREFIX, CN04_ERROR_PREFIX_DATABASE| 2)
+ERROR-PREFIX-DATABASE-5  | CN04_ERROR_PREFIX_DATABASE                                                        | 2)
+ERROR-PREFIX-DATABASE-6  | CN04_IPV4_DIFFERENT_PREFIX, CN04_IPV6_DIFFERENT_PREFIX, CN04_ERROR_PREFIX_DATABASE| 2)
```

Comment:
I guess the specification should be set first.

Reply:
In this case it should still be CN04_ERROR_PREFIX_DATABASE according to the specification.

Reply:
No, because one IPv4 address belongs to some other prefix (but it failed to find out what),
```
   3. If all members of *NS IPv4* are members of the same IP prefix in
      *IPv4 Prefix* then output *[CN04_IPV4_SINGLE_PREFIX]*.
```
error-prefix-database-6.connectivity04.xa has two NS IPv4, 127.100.101.1 and 127.100.125.1.  127.100.101.1 belongs to 127.100.101.0/24 and 127.100.125.1 belongs to some other, undetermined prefix.

And the equivalent for IPv6.